src/mito2/src/worker/handle_open.rs

```diff
@@ -71,18 +86,33 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             self.intermediate_manager.clone(),
         )
         .skip_wal_replay(request.skip_wal_replay)
-        .parse_options(request.options)?
-        .cache(Some(self.cache_manager.clone()))
-        .open(&self.config, &self.wal)
-        .await?;
-
-        info!("Region {} is opened", region_id);
+        .parse_options(request.options)
+        {
+            Ok(opener) => opener.cache(Some(self.cache_manager.clone())),
```

Comment:
You could move `.cache(Some(self.cache_manager.clone()))` in front of `parse_options()`.